scene/gui/control.h

```diff
@@ -166,46 +166,46 @@ class Control : public CanvasItem {
 		Point2 pos_cache;
 		Size2 size_cache;
 		Size2 minimum_size_cache;
-		bool minimum_size_valid;
+		bool minimum_size_valid = false;
 
 		Size2 last_minimum_size;
-		bool updating_last_minimum_size;
+		bool updating_last_minimum_size = false;
 
-		float margin[4];
-		float anchor[4];
-		FocusMode focus_mode;
-		GrowDirection h_grow;
-		GrowDirection v_grow;
+		float margin[4] = {0};
+		float anchor[4] = {ANCHOR_BEGIN};
```

Comment:
I'm not sure, but this probably initialize anchors with values {ANCHOR_BEGIN,0,0,0}

Reply:
I've always found the = {0} convention a bit redundant, that 0 defines the value of the first element in the array and the others will already default to 0.

Reply:
> the others will already default to 0.

Are you sure that `float`s are always initialized to `0.0`? Because at least `bool` and `int` aren't and this causes lots of bugs due to uninitialized member variables.

Reply:
The point was rather that `{0}` is fully equivalent to `{}` because user every single element in array is fill with 0 when using `{}` unless user fill it with default value e.g. this are equivalent lines
```
margin[4] = {};
margin[4] = {0};
margin[4] = {0,0};
margin[4] = {0,0,0,0};
```
but of course this will allow to use non-initialized values which is a bug
```
margin[4];
```

Reply:
Thanks for the clarification :+1:

Reply:
You were right, I changed it to `{ANCHOR_BEGIN, ANCHOR_BEGIN, ANCHOR_BEGIN, ANCHOR_BEGIN}`.